src/client/src/core/util/validation/__tests__/rules.spec.ts

```diff
@@ -879,3 +880,25 @@ describe('core/util/validation/rules::validLangString', () => {
     expect(result).toStrictEqual(true);
   });
 });
+
+describe('core/util/validation/rules::validSelection', () => {
+  test('validSelection:: with property selected', () => {
+    const result = validSelection({id: 1, label: 'System User'});
+    expect(result).toStrictEqual(true);
+  });
+
+  test('validSelection:: with string value', () => {
+    const result = validSelection('system user');
+    expect(result).toStrictEqual('Invalid');
+  });
+
+  test('validSelection:: with null value', () => {
+    const result = validSelection(null);
+    expect(result).toStrictEqual(true);
+  });
+
+  test('validSelection:: with empty array value', () => {
+    const result = validSelection([]);
+    expect(result).toStrictEqual(true);
+  });
```

Comment:
any idea if we pass a number? is it catch here from TS.
e.g. `validSelection(10);`

Reply:
Yeah cached from TS. can't pass number or it will error. Also the actual logic is very simple, only checking string and there's no other way numbers passed from autocomplete.